apis/core/v1beta1/openstackcontrolplane_webhook.go

```diff
@@ -46,14 +49,40 @@ var _ webhook.Validator = &OpenStackControlPlane{}
 func (r *OpenStackControlPlane) ValidateCreate() error {
 	openstackcontrolplanelog.Info("validate create", "name", r.Name)
 
-	return r.ValidateServices()
+	var allErrs field.ErrorList
+
+	allErrs = append(allErrs, r.ValidateServices()...)
+
+	if len(allErrs) != 0 {
+		return apierrors.NewInvalid(
+			schema.GroupKind{
+				Group: "openstackcontrolplane.openstack.org",
+				Kind:  "OpenstackControlplane",
+			},
+			r.Name, allErrs)
+	}
+
+	return nil
 }
 
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
 func (r *OpenStackControlPlane) ValidateUpdate(old runtime.Object) error {
 	openstackcontrolplanelog.Info("validate update", "name", r.Name)
 
-	return r.ValidateServices()
+	var allErrs field.ErrorList
+
+	allErrs = append(allErrs, r.ValidateServices()...)
+
+	if len(allErrs) != 0 {
+		return apierrors.NewInvalid(
+			schema.GroupKind{
+				Group: "openstackcontrolplane.openstack.org",
+				Kind:  "OpenstackControlplane",
```

Comment:
I don't think it really matters bit I think the last P should be capitalized here right (to fully match the CRD)? OpenStackControlPlane

Reply:
I had to rebase, and change the MariaDB/Galera check to add its error to the list so I took the opportunity to change to capital P here as well.